### DISCUSSION TITLE
Inject @Repository interface from external Micronaut library

### DISCUSSION THREAD

**Discussion by gbennardo** [micronaut-projects/micronaut-core#10938] [General]
Hi everyone and thanks in advance for your support, I'm trying to build a Micronaut library in order to access database and to avoid repeated code in various projects. 

I'm able to Inject `@Singleton` annotatated classes from the external Micronaut library to a given project but using `@Import` annotation and `micronaut-inject-java` annotation processor. I also tryed to use `micronaut-data-processor` as annotation processor but nothing changes: the `@Repository` annotated class cannot be injected. 

I'm able to inject `@Repository` annotated class into Application but not into a `@MicronautTest` class. What am I missing?

**Comment by yawkat** [comment]
@dstepanov

**Reply by dstepanov** [reply]
Duplicate of https://github.com/micronaut-projects/micronaut-data/discussions/3011
